test/e2e/e2e_test.go

```diff
@@ -863,3 +866,96 @@ var _ = Describe("e2e multiple external gateway update validation", func() {
 		}
 	})
 })
+
+func checkPodSleep(f *framework.Framework, nodeName, podName string, errChan chan error) {
+	contName := fmt.Sprintf("%s-container", podName)
+
+	command := []string{
+		"bash", "-c",
+		"sleep 10",
+	}
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:    contName,
+					Image:   framework.AgnHostImage,
+					Command: command,
+				},
+			},
+			NodeName:      nodeName,
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+	podClient := f.ClientSet.CoreV1().Pods(f.Namespace.Name)
+	result, err := podClient.Create(pod)
+	if err != nil {
+		errChan <- err
+		return
+	}
+
+	framework.Logf("Created pod %q with IP %q %q", result.GetName(), result.Status.PodIPs, result.Status.PodIPs)
+
+	err = e2epod.WaitForPodNotPending(f.ClientSet, f.Namespace.Name, podName)
+	if err != nil {
+		errChan <- err
+		return
+	}
+
+	err = e2epod.WaitForPodSuccessInNamespace(f.ClientSet, podName, f.Namespace.Name)
+
+	if err != nil {
+		logs, logErr := e2epod.GetPodLogs(f.ClientSet, f.Namespace.Name, pod.Name, contName)
+		if logErr != nil {
+			framework.Logf("Warning: Failed to get logs from pod %q: %v", pod.Name, logErr)
+		} else {
+			framework.Logf("pod %s/%s logs:\n%s", f.Namespace.Name, pod.Name, logs)
+		}
+	}
+
+	errChan <- err
+}
+
+var _ = Describe("e2e longevity test", func() {
+	var svcname = "nettest"
+
+	f := framework.NewDefaultFramework(svcname)
+
+	ginkgo.It("should create pods for a while without failing", func() {
+		ginkgo.By("Continuously creating pods for a while")
+
+		pod := 0
+		startTime := time.Now()
+
+		for time.Since(startTime) < 30*time.Minute {
+			// spin up 20 containers that each do nothing for 30 seconds
+
+			var errChans [10]chan error
+			for i, _ := range errChans {
+				errChans[i] = make(chan error)
+				go checkPodSleep(f, "", fmt.Sprintf("test-container-%v", pod), errChans[i])
```

Comment:
Short history first, 
when using go routines inside Ginkgo you should add the following statement inside of the goroutine to avoid panics
`defer GinkgoRecover()`

xref http://tonyyang132.blogspot.com/2018/08/ginkgo-and-goroutine.html